cartridge/roles/ddl-manager.lua

```diff
@@ -0,0 +1,283 @@
+        return _example_schema
+    else
+        return schema_yml
+    end
+end
+
+--- Get clusterwide schema as a Lua table.
+--
+-- In case there's no schema set, return empty schema `{spaces = {}}`.
+--
+-- @function get_clusterwide_schema_lua
+-- @treturn table schema
+local function get_clusterwide_schema_lua()
+    local schema_yml = cartridge.config_get_readonly(_section_name)
+    local schema_lua = schema_yml and yaml.decode(schema_yml)
+    if schema_lua == nil then
+        return {spaces = {}}
+    else
+        return schema_lua
+    end
+end
+
+--- Apply schema (as a YAML string) on a whole cluster.
+--
+-- @function set_clusterwide_schema_yaml
+-- @tparam string schema_yml
+-- @treturn[1] boolean `true`
+-- @treturn[2] nil
+-- @treturn[2] table Error object
+local function set_clusterwide_schema_yaml(schema_yml)
+    local patch
+    if schema_yml == nil then
+        patch = {[_section_name] = box.NULL}
+    elseif type(schema_yml) ~= 'string' then
+        local err = string.format(
+            'Bad argument #1 to set_clusterwide_schema_yaml' ..
+            ' (?string expected, got %s)', type(schema_yml)
+        )
+        error(err, 2)
+    else
+        patch = {[_section_name] = schema_yml}
+    end
+
+    return cartridge.config_patch_clusterwide(patch)
+end
+
+--- Apply schema (as a Lua table) on a whole cluster.
+--
+-- @function set_clusterwide_schema_lua
+-- @tparam string schema_lua
+-- @treturn[1] boolean `true`
+-- @treturn[2] nil
+-- @treturn[2] table Error object
+local function set_clusterwide_schema_lua(schema_lua)
+    local patch
+    if schema_lua == nil then
+        patch = {[_section_name] = box.NULL}
+    elseif type(schema_lua) ~= 'table' then
+        local err = string.format(
+            'Bad argument #1 to set_clusterwide_schema_lua' ..
+            ' (?table expected, got %s)', type(schema_lua)
+        )
+        error(err, 2)
+    else
+        patch = {[_section_name] = yaml.encode(schema_lua)}
+    end
+
+    return cartridge.config_patch_clusterwide(patch)
+end
+
+--- Validate schema passed as a YAML string.
+--
+-- @function check_schema_yaml
+-- @tparam string schema_yml
+-- @treturn[1] boolean `true`
+-- @treturn[2] nil
+-- @treturn[2] table Error object
+local function check_schema_yaml(schema_yml)
+    if schema_yml == nil then
+        return true
+    elseif type(schema_yml) ~= 'string' then
+        return nil, CheckSchemaError:new(
+            'Bad argument #1 to check_schema_yaml' ..
+            ' (?string expected, got %s)', type(schema_yml)
+        )
+    end
+
+    local schema_lua, err = DecodeYamlError:pcall(yaml.decode, schema_yml)
+    if err ~= nil then
+        return nil, err
+    elseif schema_lua == nil then
+        return true
+    end
+
+    if not failover.is_leader() then
+        return cartridge.rpc_call(
+            'ddl-manager', 'check_schema_yaml', {schema_yml},
+            {prefer_local = true, leader_only = true}
```

Comment:
Current instance is not a leader why do you need "prefer_local = true" here?

Reply:
Fixed.